Strange massive false positive case

After updating `8.5.0` -> `8.6.0`, in the `core-js` repo I see

<img width="935" alt="Image" src="https://github.com/user-attachments/assets/ab772e8c-2e73-4690-9594-d109de4b4ec9" />

For example, on

```js
var array = [];
var constructor = array.constructor = {};
```

I see

```
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-every
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-filter
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-foreach
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-indexof
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-lastindexof
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-map
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-reduce
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-reduceright
  2:19  error  ES5 'Array.prototype.constructor' method is forbidden     es-x/no-array-prototype-some
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-copywithin
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-entries
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-fill
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-find
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-findindex
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-keys
  2:19  error  ES2015 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-values
  2:19  error  ES2016 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-includes
  2:19  error  ES2019 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-flat
  2:19  error  ES2022 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-at
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es-x/no-array-prototype-findlast-findlastindex
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-toreversed
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-tosorted
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-tospliced
  2:19  error  ES2023 'Array.prototype.constructor' method is forbidden  es/no-array-prototype-with
```

--- a/lib/util/define-prototype-properties-handler/index.js
+++ b/lib/util/define-prototype-properties-handler/index.js
@@ -83,7 +83,10 @@ function definePrototypePropertiesHandler(context, propertyTypeMap, options) {
         propertyTypeMap,
     })
 
-    const propertyEntries = Object.entries(propertyTypeMap)
+    const propertyEntries = Object.entries(propertyTypeMap).map(
+        ([className, properties]) =>
+            /** @type {const} */ ([className, Object.keys(properties)]),
+    )
 
     /**
      * @param {MemberExpression|Property} node
@@ -126,7 +129,7 @@ function definePrototypePropertiesHandler(context, propertyTypeMap, options) {
     // eslint-disable-next-line func-style
     let verifyPropertyName = (node, propertyName, objectNode) => {
         for (const [className, properties] of propertyEntries) {
-            if (!properties[propertyName]) {
+            if (!properties.includes(propertyName)) {
                 continue
             }
             if (
@@ -145,7 +148,7 @@ function definePrototypePropertiesHandler(context, propertyTypeMap, options) {
         // For performance
         const [[className, properties]] = propertyEntries
         verifyPropertyName = (node, propertyName, objectNode) => {
-            if (!properties[propertyName]) {
+            if (!properties.includes(propertyName)) {
                 return
             }
             verifyClassPropertyName(node, className, propertyName, objectNode)

--- a/tests/lib/rules/no-array-prototype-at.js
+++ b/tests/lib/rules/no-array-prototype-at.js
@@ -23,6 +23,8 @@ new RuleTester().run(ruleId, rule, {
         },
         // Test for https://github.com/eslint-community/eslint-plugin-es-x/issues/223
         "for (const { x } of foo) {}",
+        `var array = [];
+         var constructor = array.constructor = {};`,
     ],
     invalid: [
         {

--- a/tests/lib/rules/no-array-prototype-every.js
+++ b/tests/lib/rules/no-array-prototype-every.js
@@ -21,6 +21,8 @@ new RuleTester().run(ruleId, rule, {
             options: [{ aggressive: false }],
             settings: { "es-x": { aggressive: true } },
         },
+        `var array = [];
+         var constructor = array.constructor = {};`,
     ],
     invalid: [
         {

